### PR TITLE
Update node image to latest LTS, version 16

### LINKED ISSUE #63: node version 10 is a legacy version! We recommend upgrading.
When installing Lando, I'm seeing 

`node version 10 is a legacy version! We recommend upgrading.`

We should update it. 

I'm not sure if we actually need node, these days, it was used before to compile scss into css and other front-end build processes.

### PR DESCRIPTION
**Pull request by finnlewis**

### CODE REVIEW

**Comment by finnlewis** [comment]
Hey @stephen-cox 

I made some changes to the Github workflow configuration to get the testing setup to install and pass, based on recent updates on the localgov profile test config (https://github.com/localgovdrupal/localgov/blob/2.x/.github/workflows/test.yml) 

Could you check it looks about right? 

Many thanks,

Finn

**Comment by andybroomfield** [comment]
Getting this error doing a lando rebuild
````
Unhandled rejection Error: node version 16 is not supported
    at new LandoCompose (/snapshot/cli/plugins/lando-core/compose/lando/builder.js)
    at new LandoAppserver (/snapshot/cli/plugins/lando-core/types/appserver/builder.js)
    at new LandoNode (/snapshot/cli/plugins/lando-services/services/node/builder.js)
    at /snapshot/cli/plugins/lando-services/app.js
    at arrayEach (/snapshot/cli/node_modules/lodash/lodash.js:530:11)
    at Function.forEach (/snapshot/cli/node_modules/lodash/lodash.js:9410:14)
    at AsyncEvents.<anonymous> (/snapshot/cli/plugins/lando-services/app.js)
    at AsyncEvents.handle (/snapshot/cli/lib/events.js)
    at /snapshot/cli/lib/events.js
From previous event:
    at AsyncEvents.emit (/snapshot/cli/lib/events.js)
    at /snapshot/cli/lib/app.js
From previous event:
    at App.init (/snapshot/cli/lib/app.js)
    at /snapshot/cli/bin/lando.js
From previous event:
    at Object.<anonymous> (/snapshot/cli/bin/lando.js)
    at Module._compile (pkg/prelude/bootstrap.js:1751:22)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:14)
    at Function.runMain (pkg/prelude/bootstrap.js:1804:12)
    at internal/main/run_main_module.js:17:47
````

**Review by andybroomfield — APPROVED** [review]
Ok, error was not using Lando 3.6
Having done that, node 16 pull through ok. I did a test with `lando yarn install` and that works so I guess its fine. Maybe @cjstevens78 might want to weigh in.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -18,31 +18,45 @@ jobs:
...
+      - name: Set composer branch reference for non-version branches
         if: endsWith(github.ref, '.x') == false
         run: echo "COMPOSER_REF=dev-${GIT_BRANCH}" >> $GITHUB_ENV
-      
+
+      - name: Get the latest tagged release for branch version
+        run: |
+          LATEST_RELEASE=$(curl -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/${GITHUB_REPOSITORY}/git/matching-refs/tags/${GIT_BASE%'.x'} | grep -Po '(?<=refs/tags/)[^"]+' | tail -1)
+          if [ -z $LATEST_RELEASE ]; then LATEST_RELEASE=1; fi
+          echo "LATEST_RELEASE=${LATEST_RELEASE}" >> $GITHUB_ENV
```
Comment (stephen-cox) [inline .github/workflows/test.yml]: This isn't needed to run tests on the project.

**Review by stephen-cox — COMMENTED** [review]
Otherwise looks fine.

**Pull request merged**